java/client/src/main/java/glide/api/models/commands/RedisScoreLimit.java

```diff
@@ -0,0 +1,50 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands;
+
+public class RedisScoreLimit {
+    public interface ScoreLimit {
+        String toArg();
+    }
+
+    /** Enumeration representing numeric positive and negative infinity bounds for a sorted set. */
+    public enum InfBound implements ScoreLimit {
```

Comment:
You can remove the `InfBound` completely if we use a `bound` of type `Double`.  Then your toArgs() call becomes: 
```
        public final static String REDIS_POSITIVE_INFINITY_ARG = "+inf";
        public final static String REDIS_NEGATIVE_INFINITY_ARG = "+inf";

        public String toArg() {
            if (this.bound == Double.POSITIVE_INFINITY) {
                return REDIS_POSITIVE_INFINITY_ARG;
            }
            if (this.bound == Double.NEGATIVE_INFINITY) {
                return REDIS_NEGATIVE_INFINITY_ARG;
            }
            return this.isInclusive ? String.valueOf(this.bound) : "(" + this.bound;
        }
```

Reply:
The advantage of this, is that we reuse existing java components, and a users don't need to check their Double values for positive/negative infinity before parsing to a double.

Reply:
I don't agree with this firstly the issue I am trying to solve with the InfBound class is the unions in python they pass a union of [InfBound, ScoreBoundry] and I solved the issue by making those two classes implement the ScoreLimit Interface. Secondly python and node also have infinite values but they used their own enums to represent it hence for consistency I think InfBound should stay.

Reply:
Ok - let's just add an extra code to the ScoreBoundary.toArgs() to check for `Double.POSITIVE_INFINITY` and `Double.NEGATIVE_INFINITY`. Otherwise, we won't process these values correctly.